src/Processors/QueryPlan/JoinStep.cpp

```diff
@@ -168,7 +168,7 @@ QueryPipelineBuilderPtr JoinStep::updatePipeline(QueryPipelineBuilders pipelines
         });
     }
 
-    if (join->supportParallelJoin())
+    if (join->supportParallelJoin() && (min_block_size_rows > 1 || min_block_size_bytes > 1))
```

Comment:
I don't see why we need this change. If limits are unset, squashing will be a no-op; simply forward all blocks further. 

https://github.com/ClickHouse/ClickHouse/blob/9fca082792b2dfbfc609e07cce51c20729b686a4/src/Interpreters/Squashing.cpp#L193-L195